React/Views/RCTShadowView.m

```diff
@@ -86,6 +86,28 @@ static void RCTProcessMetaProps(const float metaProps[META_PROP_COUNT], float st
   : 0;
 }
 
+/*
+ * Sets the measure function on the cssNode, when the
+ * aspectRatioBox rectangle is set. This helps calculate
+ * the height of the node based on width of parent and
+ * maintains the aspect ratio of the node.
+ */
+static css_dim_t RCTAspectRatioMeasure(void *context, float width)
+{
+    RCTShadowView *shadowView = (__bridge RCTShadowView *)context;
+    if (isnan(width)) width = shadowView.aspectRatioBox.width;
+    
+    CGFloat computedHeight = 0.0f;
+    if (shadowView.aspectRatioBox.width > 0) {
+        computedHeight = width * (shadowView.aspectRatioBox.height / shadowView.aspectRatioBox.width);
+    }
+    
+    css_dim_t result;
+    result.dimensions[CSS_WIDTH] = ceilf(width);
```

Comment:
Please don't round here. We only want to round exactly once when we apply it to the view, otherwise we're losing precision all along the way.

And, if we were to round here, we would need to ask many questions like
- why ceil instead of floor or round?
- the pixel density needs to be taken into account. `1` unit is equivalent to `2` physical pixels on iphone 5 for example. If you were to do this, you would lose half your precision